go/private/rules/transition.bzl

```diff
@@ -366,6 +398,39 @@ go_transition.
 """,
 )
 
+def _non_go_transition_impl(settings, attr):
+    """Sets all Go settings to the values they had before the last go_transition.
+
+    non_go_transition sets all of the //go/config settings to the value they had
+    before the last go_transition. This should be used on all attributes of
+    go_library/go_binary/go_test that are built in the target configuration and
+    do not constitute advertise any Go providers.
+
+    Examples: This transition is applied to the 'data' attribute of go_binary so
+    that other Go binaries used at runtime aren't affected by a non-standard
+    link mode set on the go_binary target, but still use the same top-level
+    settings such as e.g. race instrumentation.
+    """
+    new_settings = {}
+    for key in TRANSITIONED_GO_SETTING_KEYS:
+        original_key = _original_setting_key(key)
+        original_value = settings[original_key]
```

Comment:
I could implement that change, but could you explain why you think it's needed? If `_original_setting_key(key)` is `None` for `key` in `TRANSITIONED_GO_SETTING_KEYS`, then that's a bug in this file. I could possibly see adding a check with a `fail`, but a silent `continue` would - imo - just make these bugs harder to spot.

Reply:
yeah, adding a `fail` with more debuggable message is better. Because `_original_setting_key(key)` can return `None`, it's better to check for that when `None` is not acceptable.

Reply:
I added a commit that pushed the conversion into the construction of a dict that contains the mapping and added a fail in the conversion function. What do you think?